Eatery/Extensions/Eatery+Extension.swift

```diff
@@ -0,0 +1,176 @@
+            case .open: return ["Open", "Closed"]
+            case .paymentType: return ["Swipes", "BRB", "Cash"]
+            }
+        }
+
+        var sectionCount: Int {
+            return self.names.count
+        }
+    }
+
+    //!TODO: Maybe cache this value? I don't think this is too expensive
+    var favorite: Bool {
+        get {
+            let ar = UserDefaults.standard.stringArray(forKey: "favorites") ?? []
+            return ar.contains {
+                $0 == slug
+            }
+        }
+
+        set {
+            var ar = UserDefaults.standard.stringArray(forKey: "favorites") ?? []
+            let contains = self.favorite
+            if (newValue && !contains) {
+                ar.append(self.slug)
+            } else if (!newValue && contains) {
+                let idx = ar.index {
+                    $0 == slug
+                }
+
+                if let idx = idx {
+                    ar.remove(at: idx)
+                }
+            }
+
+            UserDefaults.standard.set(ar, forKey: "favorites");
+        }
+    }
+
+    // ** COPY OF IMPLEMENTATION IN LiteEatery.swift
+    // TODO: refactor to avoid repeated code
+    //
+    // Generates description of eatery for its current state
+    // returns "Opening in x min)" if x <= 60 and is closed"
+    // "Closing in x min) if x <= 60 and is open,
+    // "Closed" if closed and not opening soon,
+    // "Open now" if open and not closing soon
+    // Bool value is either stable or about to change
+    func generateDescriptionOfCurrentState() -> EateryStatus {
+        if isOpenToday() {
+            guard let activeEvent = activeEventForDate(Date()) else {
+                return .closed("Closed", "")
+            }
+
+            if activeEvent.occurringOnDate(Date()) {
+                let minutesTillClose = (Int)(activeEvent.endDate.timeIntervalSinceNow/Double(60))
+                if minutesTillClose < 30 {
+                    return .closing("Closing", "in \(minutesTillClose+1)m")
+                } else {
+                    let timeString = ShortDateFormatter.string(from: activeEvent.endDate)
+                    return .open("Open", "until \(timeString)")
+                }
+            } else {
+                let minutesTillOpen = (Int)(activeEvent.startDate.timeIntervalSinceNow/Double(60))
+                if minutesTillOpen < 60 {
+                    return .closed("Opening", "in \(minutesTillOpen+1)m")
+                } else {
+                    let timeString = ShortDateFormatter.string(from: activeEvent.startDate)
+                    return .closed("Closed", "until \(timeString)")
+                }
+            }
+        } else {
+            return .closed("Closed", "today")
+        }
+    }
+
+    // Retrieves a string list of the hours of operation for a day/time
+    func activeEventsForDate(date: Date) -> String {
+        var resultString = "Closed"
```

Comment:
Be more explicit about when this is "Closed". `if events.isEmpty`, else as such.